src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedFieldSymbolBase.cs

```diff
@@ -45,13 +45,10 @@ internal override void AddSynthesizedAttributes(ModuleCompilationState compilati
 
             CSharpCompilation compilation = this.DeclaringCompilation;
 
-            // Assume that someone checked earlier that the attribute ctor is available and has no use-site errors.
             AddSynthesizedAttribute(ref attributes, compilation.TrySynthesizeAttribute(WellKnownMember.System_Runtime_CompilerServices_CompilerGeneratedAttribute__ctor));
 
-            // TODO (tomat): do we need to emit dynamic attribute on any synthesized field?
-            if (this.Type.ContainsDynamic())
+            if (this.Type.ContainsDynamic() && compilation.HasDynamicEmitAttributes())
```

Comment:
This check is not complete, see similar method in SynthesizedParameterSymbol, it also checks goodness of a boolean type. Perhaps that check should be moved to HasDynamicEmitAttributes.